lib/reline/line_editor.rb

```diff
@@ -275,40 +207,27 @@ def reset_variables(prompt = '', encoding:)
     @first_prompt = true
     @searching_prompt = nil
     @first_char = true
-    @add_newline_to_end_of_buffer = false
-    @just_cursor_moving = nil
```

Comment:
`@just_cursor_moving`, `@scroll_partial_screen` are still using. Is it safe to delete?

Reply:
`@scroll_partial_screen` is not deleted.
```diff
- @scroll_partial_screen = nil
+ @scroll_partial_screen = 0
```

I restored `@just_cursor_moving = false`. thanks